rust/kernel/delay.rs

```diff
@@ -56,6 +56,33 @@ pub fn coarse_sleep(duration: Duration) {
     unsafe { bindings::msleep(coarse_sleep_conversion(duration)) }
 }
 
+/// busy wait for enough loop cycles to achieve the desired delay.
+///
+/// This function supports the C side `mdelay`, `udelay`, and `ndelay` functions.
+pub fn coarse_delay(duration: Duration) {
+    let millis = min(
+        duration
+            .as_secs()
+            .saturating_mul(1_000)
+            .saturating_add(duration.subsec_millis() as u64) as core::ffi::c_ulong,
+        core::ffi::c_ulong::MAX,
+    );
```

Comment:
Does `min` do anything here? `MAX` can never be less than the other argument, since they're the same width.

Reply:
Yeah, looks like useless. I can't recall why I put min() here... I removed it.